src/stores/store/database/postgres/keys.go

```diff
@@ -0,0 +1,121 @@
+package postgres
+
+import (
+	"context"
+	"time"
+
+	"github.com/consensys/quorum-key-manager/src/infra/log"
+	"github.com/consensys/quorum-key-manager/src/infra/postgres"
+
+	"github.com/consensys/quorum-key-manager/src/stores/store/database"
+	"github.com/consensys/quorum-key-manager/src/stores/store/entities"
+)
+
+type Keys struct {
+	logger log.Logger
+	db     postgres.Client
+}
+
+var _ database.Keys = &Keys{}
+
+func NewKeys(logger log.Logger, db postgres.Client) *Keys {
+	return &Keys{
+		logger: logger,
+		db:     db,
+	}
+}
+
+func (d *Keys) Get(_ context.Context, id string) (*entities.Key, error) {
+	key := &entities.Key{ID: id}
+
+	err := d.db.SelectPK(key)
+	if err != nil {
+		d.logger.WithError(err).Error("failed to get key")
+		return nil, err
+	}
+
+	return key, nil
+}
+
+func (d *Keys) GetDeleted(_ context.Context, id string) (*entities.Key, error) {
+	key := &entities.Key{ID: id}
+
+	err := d.db.SelectDeletedPK(key)
```

Comment:
Every call to DB should depend on ctx otherwise if the routine making the call went death we make DB compute work for not need. Same it would be for any call to external dependencies, context should define the lifetime of the call

Reply:
are you sure about that? Because AKV and AWS do not accept context if I'm not mistaken. Also go-pg examples do not use the context.

Reply:
added